### ISSUE TITLE
Linear Regression

### ISSUE THREAD

**Issue by phymo** [phymooc/learn-AI#5]
- 似然函数 
- 梯度下降：gradient descent
  1. 批量梯度下降
  2. 随机梯度下降
  3. 小批量梯度下降
- normalization
- 特征值 features
- 目的：计算参数 theta 向量
- 最小二乘法

**Comment by phymo** [comment]
Linear Regression Class

**Comment by phymo** [comment]
Trainning:
1. dataset: training data and test data

**Comment by phymo** [comment]
为什么线性（或者说一阶直线）：高阶拟合容易过拟合

**Comment by phymo** [comment]
直接可以计算出 theta? 那怎么做训练呢？ => 梯度下降
